### PR TITLE
Cherry pick NavigationBar ink ripple fixes

### PR DESCRIPTION
**Pull request by Piinks**
Pick for: https://github.com/flutter/flutter/issues/119554

Main goal to cherry pick: https://github.com/flutter/flutter/pull/117473
Required pulling in already cherry picked: https://github.com/flutter/flutter/pull/117500

There were some conflicts based on a lot of changes in this file over the last month, some cherry picked, others not, can you check me on this @TahaTesser?

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by TahaTesser** [comment]
@Piinks 
This looks alright to me.

**Review by TahaTesser — APPROVED** [review]

**Comment by XilaiZhang** [comment]
Umm I did a rerun of the presubmit tests but seem like they are still failing

**Comment by Piinks** [comment]
Yeah I am not sure what the issue is, there are a number of errors that have to do with unnamed library?

```
01:00 +1000 ~1: /b/s/w/ir/x/w/flutter/packages/flutter/test/material/calendar_date_picker_test.dart: CalendarDatePicker Can select a day                                                               
test/material/navigation_bar_theme_test.dart:8:8: Error: The 'unnamed-libraries' language feature is disabled for this library.
Try removing the package language version or setting the language version to 2.19 or higher.
library;
```

I am not used to cherry picking, I'll try to take another look later today.

File: packages/flutter/test/material/navigation_bar_test.dart [hunk 1]
```diff
@@ -2,6 +2,11 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+// This file is run as part of a reduced test set in CI on Mac and Windows
+// machines.
+@Tags(<String>['reduced-test-set'])
+library;
```
Comment (goderbauer) [inline packages/flutter/test/material/navigation_bar_test.dart:8]: Regarding the error: You probably have to delete this line. The empty library statement is a new Dart feature that is likely not available with the dart version used by the stable branch.

```suggestion
```

File: packages/flutter/test/material/navigation_bar_theme_test.dart [hunk 2]
```diff
@@ -2,6 +2,12 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+// This file is run as part of a reduced test set in CI on Mac and Windows
+// machines.
+@Tags(<String>['reduced-test-set'])
+library;
```
Comment (goderbauer) [inline packages/flutter/test/material/navigation_bar_theme_test.dart:8]: Same:

```suggestion
```

**Comment by XilaiZhang** [comment]
We will probably save this cp for the next hotfix, since we are about to release tomorrow morning

**Comment by Piinks** [comment]
Thanks, should be fixed with latest commit

**Comment by godofredoc** [comment]
Landing with customer_testing failures as they are not expected to pass consistently on release candidate branches.

**Pull request merged**